File: fullstack-online/unit_09/cryptipedia/client/src/models/Cryptid.js
import { Profile } from "./Account.js"

export class Cryptid {
  constructor(data) {
    this.id = data.id
    this.createdAt = new Date(data.createdAt)
    this.updatedAt = new Date(data.updatedAt)
    this.name = data.name
    this.threatLevel = data.threatLevel
    this.imgUrl = data.imgUrl
    this.origin = data.origin
    this.size = data.size
    this.description = data.description
    this.discovererId = data.discovererId
    this.discoverer = new Profile(data.discoverer)
    this.encounterCount = data.encounterCount
  }
}

File: fullstack-online/unit_09/cryptipedia/server/Models/Cryptid.cs
using System.ComponentModel.DataAnnotations;

namespace cryptipedia.Models;

public class Cryptid : RepoItem<int>
{
  public string Name { get; set; }
  [Range(0, 10)] public int ThreatLevel { get; set; }
  [Url] public string ImgUrl { get; set; }
  public string Origin { get; set; }
  [Range(0, 10)] public int Size { get; set; }
  [MaxLength(10000)] public string Description { get; set; }
  public string DiscovererId { get; set; }
  public Profile Discoverer { get; set; }
  public int EncounterCount { get; set; }
}